### PR TITLE
Using node location to track debug points

### PR DESCRIPTION
**Pull request by pahans**

### CODE REVIEW

File: modules/web/js/ballerina/views/ballerina-file-editor.js [hunk 1]
```diff
@@ -1070,27 +1065,81 @@ class BallerinaFileEditor extends BallerinaView {
...
+        pathVectors.forEach(pathVectorStr => {
+            const pathVector = JSON.parse(pathVectorStr);
+            const node = this.getNodeByVector(this._model, pathVector);
+            const lineNumber = node.getLineNumber();
+            breakpoints.push(lineNumber);
+        });
+        this._sourceView.setBreakpoints(breakpoints);
+    }
+
+    hideCurrentBreakpoints() {
```
Comment (hemikak) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: Need to have doc comments.

File: modules/web/js/ballerina/views/ballerina-file-editor.js [hunk 2]
```diff
@@ -1070,27 +1065,81 @@ class BallerinaFileEditor extends BallerinaView {
...
+            nodeView.showDebugIndicator();
+            // DebugManager.addBreakPoint(breakpointNode.getLineNumber(), fileName);
+            const pathVector = [];
+            this.getPathToNode(breakpointNode, pathVector);
+            this._currentBreakpoints.push(JSON.stringify(pathVector));
         });
     }
 
-    _hideBreakpoint() {
+    _showSourceViewBreakPoints() {
```
Comment (hemikak) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: Need to have doc comments.

File: modules/web/js/ballerina/visitors/find-breakpoints-visitor.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import ASTVisitor from './ast-visitor';
+
+class FindBreakpointsVisitor extends ASTVisitor {
```
Comment (hemikak) [inline modules/web/js/ballerina/visitors/find-breakpoints-visitor.js]: Need to have a class comment.

Comment (pahans) [inline modules/web/js/ballerina/visitors/find-breakpoints-visitor.js]: fixed with https://github.com/ballerinalang/composer/pull/2011

File: modules/web/js/ballerina/views/ballerina-file-editor.js (hunk 2 above)
Comment (pahans) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: fixed in https://github.com/ballerinalang/composer/commit/816f10086d7ab68eaa48db0862e8a9f48986c8c1

File: modules/web/js/ballerina/views/ballerina-file-editor.js (hunk 1 above)
Comment (pahans) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: fixed in https://github.com/ballerinalang/composer/commit/816f10086d7ab68eaa48db0862e8a9f48986c8c1

File: modules/web/js/ballerina/views/ballerina-file-editor.js [hunk 4]
```diff
@@ -515,26 +509,27 @@ class BallerinaFileEditor extends BallerinaView {
...
+            lastRenderedTimestamp = this._file.getLastPersisted();
+            this.toolPalette.hide();
             // If the file has changed we will add the generated source to source view
             // If not we will display the content as it is in the file.
-            if(self._file.isDirty()){
-                var generatedSource = self.generateSource();
-                self._sourceView.setContent(generatedSource);
-                self._sourceView.format(true);
+            const generatedSource = this.generateSource();
+//            var model = ballerinaFileEditor.getModelFromSource(generatedSource);
```
Comment (dakshika) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: Remove commented line

File: modules/web/js/ballerina/views/ballerina-file-editor.js [hunk 5]
```diff
@@ -1070,27 +1065,81 @@ class BallerinaFileEditor extends BallerinaView {
...
+        findBreakpointsVisitor.setBreakpoints(breakpoints);
+
+        var generatedSource = this.getContent();
+        var model = this.getModelFromSource(generatedSource);
+        this.setModel(model);
+        this.reDraw();
+        const modelMap = this.diagramRenderingContext.getViewModelMap();
+        model.accept(findBreakpointsVisitor);
+        const breakpointNodes = findBreakpointsVisitor.getBreakpointNodes();
+        // const fileName = this._file.getName();
```
Comment (dakshika) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: remove commented lines

File: modules/web/js/ballerina/views/ballerina-file-editor.js (hunk 4 above)
Comment (pahans) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: this was removed during the eslint refactoring

File: modules/web/js/ballerina/views/ballerina-file-editor.js (hunk 5 above)
Comment (pahans) [inline modules/web/js/ballerina/views/ballerina-file-editor.js]: this was removed during the eslint refactoring

**Pull request merged**